### DISCUSSION TITLE
Label position when text is rotated + Show all labels

### DISCUSSION THREAD

**Discussion by jbiason** [plotters-rs/plotters#466] [Q&A]
I'm trying to use font rotation ('cause I have lots of labels), but I can't find a way to set it that the labels appear under the graph: they appear over the bar.

![cases](https://user-images.githubusercontent.com/228850/233118314-1b6f88e2-041c-455c-9f09-a56869bd037f.png)

Also, I still haven't found a way to display all labels, even setting the maximum number of labels to the the same value as the number of points.

Code is here: https://github.com/jbiason/random/blob/master/plotterstest/src/main.rs

**Comment by jbiason** [comment]
Found one thing about the last point: Apparently, to plot every single item in the X axis, Plotters need to have an empty label at the end.

So, basically, my issue is that my `.x_label_formatter()` would request 1 more item than what I have in my data. Returning an empty string for the last element fixes that issue.

**Comment by seanaye** [comment]
Did you ever find a solution to this? I'm having the same problem

**Reply by wangjiawen2013** [reply]
Could you try `("sans-serif", 20).into_font().pos()` to adust the text anchor position ?

**Comment by AliMMehr** [comment]
I just tested this in the `next-release-devel` branch and it is fixed in that branch. I mean you will need to change your [code at line 179](https://github.com/jbiason/random/blob/fe3b2122cdfa82813d2b40193ce6e33a88b2c1ef/plotterstest/src/main.rs#L179) with this pos: 
`let pos = Pos::new(HPos::Right, VPos::Bottom);`
